token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -450,139 +452,329 @@ fn process_transfer(
     let mint_info = next_account_info(account_info_iter)?;
     let instructions_sysvar_info = next_account_info(account_info_iter)?;
     let authority_info = next_account_info(account_info_iter)?;
-    let authority_info_data_len = authority_info.data_len();
 
     check_program_account(mint_info.owner)?;
     let mint_data = &mint_info.data.borrow_mut();
     let mint = StateWithExtensions::<Mint>::unpack(mint_data)?;
     let ct_mint = mint.get_extension::<ConfidentialTransferMint>()?;
 
+    let previous_instruction =
+        get_instruction_relative(proof_instruction_offset, instructions_sysvar_info)?;
+
+    if let Ok(transfer_fee_config) = mint.get_extension::<TransferFeeConfig>() {
+        // mint is extended for fees
+        let proof_data = decode_proof_instruction::<TransferWithFeeData>(
+            ProofInstruction::VerifyTransfer,
+            &previous_instruction,
+        )?;
+
+        if proof_data.transfer_with_fee_pubkeys.auditor != ct_mint.auditor_pk {
+            return Err(TokenError::ConfidentialTransferElGamalPubkeyMismatch.into());
+        }
+
+        // `withdraw_withheld_authority` ElGamal pubkey in proof data and mint must match
+        if proof_data.transfer_with_fee_pubkeys.fee_collector
+            != ct_mint.withdraw_withheld_authority_pk
```

Comment:
Same comment about the lhs and rhs fields matching for readability: `. fee_collector ` vs `. withdraw_withheld_authority_pk `

Reply:
Yep, changed `withdraw_withheld_authority_pk` to `withdraw_withheld_authority_pubkey` in the mint. Will change `fee_collctor` to the same name in the zk-token-sdk `0.7.0`.